### PR TITLE
8154038: Spinner's converter should update its editor

### PR DESCRIPTION
**Pull request by karthikpandelu**
When converter was changed, `setText` method of spinner was not invoked to update the spinner text using new converter.

Added a listener to converter property of `SpinnerValueFactory` to call `setText` so that spinner text is updated using new converter.

Added unit test to validate the fix. Along with spinner of Double type, added a spinner of Integer type as well so that the fix is validated using 2 types of spinner.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8154038](https://bugs.openjdk.org/browse/JDK-8154038): Spinner's converter should update its editor

### Reviewers
 * [Ajit Ghaisas](https://openjdk.org/census#aghaisas) (@aghaisas - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1057/head:pull/1057` \
`$ git checkout pull/1057`

Update a local copy of the PR: \
`$ git checkout pull/1057` \
`$ git pull https://git.openjdk.org/jfx.git pull/1057/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1057`

View PR using the GUI difftool: \
`$ git pr show -t 1057`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1057.diff">https://git.openjdk.org/jfx/pull/1057.diff</a>

</details>

### CODE REVIEW

**Review by aghaisas — CHANGES_REQUESTED** [review]

**Comment by andy-goryachev-oracle** [comment]
Added SpinnerPage to the MonkeyTester

https://github.com/andy-goryachev-oracle/Test/blob/main/src/goryachev/monkey/MonkeyTesterApp.java
https://github.com/andy-goryachev-oracle/Test/blob/main/src/goryachev/monkey/pages/SpinnerPage.java

File: modules/javafx.controls/src/main/java/javafx/scene/control/Spinner.java [hunk 1]
```diff
@@ -541,6 +541,10 @@ public final ReadOnlyObjectProperty<T> valueProperty() {
                         // this binding is what ensures the Spinner.valueProperty()
                         // properly represents the value in the value factory
                         value.bind(newFactory.valueProperty());
+                        // Update the spinner editor when converter is changed.
+                        newFactory.converterProperty().addListener((o, oldValue, newValue) -> {
```
[the remark this answers is not in the document]
Comment (karthikpandelu) [reply modules/javafx.controls/src/main/java/javafx/scene/control/Spinner.java]: Updated the code. Please check

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/SpinnerTest.java [hunk 2]
```diff
@@ -1523,4 +1525,56 @@ public void test_jdk_8150946_testCommit_invalid_wrongType() {
         localTimeValueFactory.setValue(null);
         assertNull(localTimeSpinner.getValue());
     }
+
+    @Test public void testSpinnerEditorUpdateOnConverterChange() {
```
[the remark this answers is not in the document]
Comment (karthikpandelu) [reply modules/javafx.controls/src/test/java/test/javafx/scene/control/SpinnerTest.java]: Created separate tests for each type of Spinner. Please check

**Review by aghaisas — APPROVED** [review]
Looks good!

**Pull request closed without merging**